crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__ruff_test__statement__try_py.snap

```diff
@@ -0,0 +1,110 @@
+---
+source: crates/ruff_python_formatter/src/lib.rs
+expression: snapshot
+---
+## Input
+```py
+try:
+    ...
+except:
+    ...
+
+
+try: # try
+    ...
+    # end of body
+# before except
+except (Exception, ValueError) as exc:  # except line
+    ...
+# before except 2
+except (KeyError) as key:  # except line 2
+    ...
+    # in body 2
+# before else
+else:
+    ...
+# before finally
+finally:
+    ...
+
+
+
+# with line breaks
+try: # try
+    ...
+    # end of body
+
+# before except
+except (Exception, ValueError) as exc:  # except line
+    ...
+
+# before except 2
+except (KeyError) as key:  # except line 2
+    ...
+    # in body 2
+
+# before else
+else:
+    ...
+
+# before finally
+finally:
+    ...
+```
+
+
+
+## Output
+```py
+try:
+    ...
+except:
+    ...
+
+
+try:
+    # try
+    ...
+    # end of body
+# before except
+except (Exception, ValueError) as exc:  # except line
+    ...
+# before except 2
+except (
+    KeyError  # except line 2
+) as key:
+    ...
+    # in body 2
+# before else
+else:
+    ...
+# before finally
+finally:
+    ...
+
+
+# with line breaks
+try:
+    # try
+    ...
+    # end of body
+# before except
```

Comment:
lots of blank lines before comments missing here. i don't recall handling that manually elsewhere but probably missing something. what's the best way to add those?

Reply:
You're already correctly using `empty_line()`, but i think you need to measure in front of the first leading comment instead of in front of the node (the tokenizer emits `TokenKind::Comment` otherwise)